### PR TITLE
Added support for the file conflicts callbacks (bnc#923590)

### PR DESCRIPTION
**Pull request by lslezak**
- 3.1.24

### CODE REVIEW

File: src/Callbacks.cc [hunk 1]
```diff
@@ -1697,6 +1698,87 @@ namespace ZyppRecipients {
...
+              callback.evaluate();
+          }
+      }
+
+      virtual bool start( const zypp::ProgressData & progress_r ) {
+          return report_progress(progress_r);
+      }
+
+      virtual bool progress( const zypp::ProgressData & progress_r,
+          const zypp::sat::Queue & noFilelist_r ) {
```
Comment (jreidinger) [inline src/Callbacks.cc]: hmm, is this indentation consistent over whole file? I find it quite bad to read as it is hard to distinguish between additional parametrs and start of body ( two problems together `{` on same line and parameters indentation same as body ).

File: src/Callbacks.cc [hunk 2]
```diff
@@ -1697,6 +1698,87 @@ namespace ZyppRecipients {
...
+          return report_progress(progress_r);
+      }
+
+      virtual bool result( const zypp::ProgressData & progress_r,
+        const zypp::sat::Queue & noFilelist_r,
+        const zypp::sat::FileConflicts & conflicts_r ) {
+          CB callback( ycpcb( YCPCallbacks::CB_FileConflictReport) );
```
Comment (jreidinger) [inline src/Callbacks.cc]: here indentation is different

File: src/Callbacks.cc [hunk 3]
```diff
@@ -1697,6 +1698,87 @@ namespace ZyppRecipients {
...
+
+          if (callback._set) {
```
Comment (jreidinger) [inline src/Callbacks.cc]: maybe use short evaluation?
```
if (!callback._set)
  // continue
  return true;

// continue without extra indentation
```

File: src/Callbacks.cc [hunk 4]
```diff
@@ -1697,6 +1698,87 @@ namespace ZyppRecipients {
...
+
+              YCPList excluded_packages;
+              for_(iter, noFilelist_r.begin(), noFilelist_r.end())
+              {
+                  // convert solvable ID to a Package
+                  zypp::Package::Ptr pkg(zypp::make<zypp::Package>(zypp::sat::Solvable(*iter)));
+                  if (pkg) {
+                      excluded_packages->add(YCPString(pkg->name() + "-" +
+                        pkg->edition().asString() + "-" + pkg->arch().asString()));
```
Comment (jreidinger) [inline src/Callbacks.cc]: there isn't any standard way in `pkg` how to get formatted output?

Comment (lslezak) [inline src/Callbacks.cc]: It seems not, and for example `zypper` [uses a stringstream for this](https://github.com/openSUSE/zypper/blob/master/src/misc.cc#L516).

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**